foundry/applications/basePlaceableHUD.d.ts

```diff
@@ -1,29 +1,91 @@
-// @TODO:
-
 /**
  * An abstract base class for displaying a heads-up-display interface bound to a Placeable Object on the canvas
  */
-declare class BasePlaceableHUD extends Application {
+declare abstract class BasePlaceableHUD<
+  P extends PlaceableObject<PlaceableObject.Data>,
+  L extends PlaceablesLayer<P>
+> extends Application {
+  constructor(...args: ConstructorParameters<typeof Application>);
+
   /**
-   * BasePlaceableHUD display states
+   * Reference a PlaceableObject this HUD is currently bound to
    */
-  DISPLAY_STATES: object;
+  object: P | null;
 
   /**
-   * Reference a PlaceableObject this HUD is currently bound to
+   * @override
+   * @defaultValue
+   * ```
+   * mergeObject(super.defaultOptions, {
+   *   classes: ["placeable-hud"],
+   *   popOut: false
+   * })
+   * ```
    */
-  object: PlaceableObject;
+  static get defaultOptions(): Application.Options;
 
-  constructor(...args: any[]);
+  /**
+   * Convenience access for the canvas layer which this HUD modifies
+   */
+  get layer(): L;
```

Comment:
Instead of making the type of the layer generic here, wouldn't it make more sense to to it in `PlaceableObject`?